test/functional/core/exit_spec.lua

```diff
@@ -0,0 +1,46 @@
+local helpers = require('test.functional.helpers')(after_each)
+
+local command = helpers.command
+local eval = helpers.eval
+local eq = helpers.eq
+local run = helpers.run
+
+describe('v:exiting', function()
+  local cid
+
+  before_each(function()
+    helpers.clear()
+    cid = helpers.nvim('get_api_info')[1]
+  end)
+
+  it('defaults to v:null', function()
+    assert(eval('v:exiting is v:null'))
```

Comment:
Argh, I commented for the wrong line. I meant to apply the change you suggested to
`assert(eval('v:exiting') ~= 0)` as well. So `neq(0, eval('v'exiting'))`. Sorry for the confusion.